### PR TITLE
fix: 403 errors pop up fix

### PR DESCRIPTION
**Pull request by nikhilsaikethe**
## Summary by CodeRabbit

- **New Features**
	- Enhanced error handling across various components to improve user feedback by selectively notifying users based on error status codes.
	- Added `data-test` attributes to several components for improved testing capabilities.

- **Bug Fixes**
	- Improved error handling logic to suppress notifications for forbidden errors (HTTP 403) in multiple components.

- **Documentation**
	- Minor formatting adjustments for better readability and consistency across components.

### CODE REVIEW

File: web/src/components/logstream/AddStream.vue [hunk 1]
```diff
@@ -238,11 +238,13 @@ const saveStream = async () => {
...
     .catch((err) => {
-      q.notify({
+      if(err.response.status != 403){
+        q.notify({
         color: "negative",
         message: err.response?.data?.message || "Failed to create stream",
         timeout: 4000,
       });
+      }
     });
```
[the remark this answers is not in the document]
Comment (nikhilsaikethe) [reply web/src/components/logstream/AddStream.vue:249]: it is working fine as of now there is no change in the code so ignoring

File: web/src/components/pipeline/PipelinesList.vue [hunk 2]
```diff
@@ -580,12 +584,14 @@ const deletePipeline = async () => {
...
     .catch((error) => {
-      q.notify({
+      if(error.response.status != 403){
+        q.notify({
         message: error.response?.data?.message || "Error while saving pipeline",
         color: "negative",
         position: "bottom",
         timeout: 3000,
       });
+      }
```
[the remark this answers is not in the document]
Comment (nikhilsaikethe) [reply web/src/components/pipeline/PipelinesList.vue:594]: yes thank you for your suggestion , fixed this

File: web/src/components/logstream/AddStream.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (nikhilsaikethe) [reply web/src/components/logstream/AddStream.vue:249]: not needed for this change so ignoring as we dont changed this

File: web/src/components/pipeline/PipelinesList.vue [hunk 3]
```diff
@@ -370,12 +370,14 @@
...
   }).catch((error) => {
-    q.notify({
+    if(error.response.status != 403){
+      q.notify({
       message: error.response?.data?.message || "Error while updating pipeline state",
       color: "negative",
       position: "bottom",
       timeout: 3000,
     });
+    }
```
[the remark this answers is not in the document]
Comment (nikhilsaikethe) [reply web/src/components/pipeline/PipelinesList.vue:380]: considered this

File: web/src/components/pipeline/PipelinesList.vue [hunk 4]
```diff
@@ -545,12 +547,14 @@ const savePipeline = (data: any) => {
...
     .catch((error) => {
-      q.notify({
+      if(error.response.status != 403){
+        q.notify({
         message: error.response?.data?.message || "Error while saving pipeline",
         color: "negative",
         position: "bottom",
         timeout: 3000,
       });
+      }
```
[the remark this answers is not in the document]
Comment (nikhilsaikethe) [reply web/src/components/pipeline/PipelinesList.vue:557]: yes removed finally block instead added dismiss() at both then and catch  blocks

File: src/infra/src/table/migration/mod.rs [hunk 5]
```diff
@@ -23,6 +23,8 @@ mod m20241119_000001_create_dashboards_table;
 mod m20241119_000002_populate_dashboards_table;
 mod m20241119_000003_delete_metas;
 mod m20241204_143100_create_table_search_queue;
+mod m20241209_120000_create_alerts_table;
+mod m20241215_190333_delete_metas;
```
[the remark this answers is not in the document]
Comment (gaby) [reply src/infra/src/table/migration/mod.rs:27]: @nikhilsaikethe @hengfeiyang This one seems important if true

**Review by bjp232004 — APPROVED** [review]

File: src/infra/src/table/migration/mod.rs (hunk 5 above)
[the remark this answers is not in the document]
Comment (hengfeiyang) [reply src/infra/src/table/migration/mod.rs]: @gaby this is a mistake by another PR. already fixed https://github.com/openobserve/openobserve/pull/5375

**Pull request closed without merging**